src/shim/preload_syscalls.c

```diff
@@ -1,401 +1,130 @@
-// Defines system call wrappers: functions that are
-// documented in man section 2. (See `man man`).
+// Defines system call wrappers: functions that are documented in man section 2. (See `man man`).
+// This file defines ths symbols that will be included in the preload library,
+// and we redirect to the syscall() function to actually handle them.
 
-#include <assert.h>
-#include <dlfcn.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <poll.h>
-#include <stdarg.h>
-#include <stdint.h>
-#include <stdlib.h>
-#include <sys/epoll.h>
-#include <sys/resource.h>
-#include <sys/select.h>
-#include <sys/stat.h>
-#include <sys/statfs.h>
+// Get the SYS_xxx definitions
 #include <sys/syscall.h>
```

Comment:
Probably worth a comment here to avoid including any other headers, which could end up conflicting with the definitions below.